docs/authoring-recipes/recipe-testing.md

```diff
@@ -321,6 +321,8 @@ In a majority of cases, though, a `SourceSpec` will also define an "after" state
 
 A developer can assert additional conditions on a source file by using the `afterRecipe` callback that is defined on the `SourceSpec`. This can be convenient when asserting conditions on the resulting semantic model that are not represented in the rendering of the source code after the recipe has transformed the source file. For instance, you may want to confirm that the path to a file has changed such as in the [sample tests](#example-tests) above. This file path is not visible in the "after" code and can only be tested via this callback.
 
+By default, the `RewriteTest` interface's `rewriteRun` method will evaluate and parse all sourceSet's "before" and "after" contents trimming the last line of the multi-line `String` blocks. To avoid this behaviour and to persist the newlines at the end of a "content" to correctly assert end-of-file cases, you can call the `noTrim` method on the `SourceSpec` to bypass this trimming. 
```

Comment:
```suggestion
By default, `RewriteTest.rewriteRun()` automatically trims trailing newlines from both the "before" and "after" source strings in your tests. This can cause issues when testing recipes that specifically handle end-of-file formatting. To preserve trailing newlines and test those edge cases, you can call the `noTrim()` method on your `SourceSpec`.
```

I found this paragraph to be a bit difficult to understand so here's a proposal I think is a bit clearer. Please let me know if I misunderstood something, though.

Reply:
Also would it be useful to add an example such as?

```
  rewriteRun(
    java(
      """
        public class Test {
        }

        """,
      """
        public class Test {
        }
        """,
      spec -> spec.noTrim()
    )
  );
```